ext/src/Duration.c

```diff
@@ -9,34 +9,31 @@
 
 zend_class_entry *php_driver_duration_ce = NULL;
 
-static void to_string(zval *result, cass_int64_t value)
+static void to_string(zval *result, cass_int32_t value)
 {
-  // Adapted from Bigint to_string function.
   char *string;
-#ifdef WIN32
-  spprintf(&string, 0, "%I64d", (long long int) value);
-#else
-  spprintf(&string, 0, "%lld", (long long int) value);
-#endif
+  spprintf(&string, 0, "%d", value);
   PHP5TO7_ZVAL_STRING(result, string);
   efree(string);
 }
 
-static int get_int64(zval* value, cass_int64_t* destination, const char* param_name TSRMLS_DC)
+static int get_int32(zval* value, cass_int32_t* destination, const char* param_name TSRMLS_DC)
 {
   // Adapted from Bigint __construct method.
   if (Z_TYPE_P(value) == IS_LONG) {
-    *destination = (cass_int64_t) Z_LVAL_P(value);
+    *destination = Z_LVAL_P(value);
   } else if (Z_TYPE_P(value) == IS_DOUBLE) {
-    *destination = (cass_int64_t) Z_DVAL_P(value);
+    *destination = (cass_int32_t) Z_DVAL_P(value);
   } else if (Z_TYPE_P(value) == IS_STRING) {
-    if (!php_driver_parse_bigint(Z_STRVAL_P(value), Z_STRLEN_P(value), destination TSRMLS_CC)) {
+    cass_int64_t parsed_big_int;
+    if (!php_driver_parse_bigint(Z_STRVAL_P(value), Z_STRLEN_P(value), &parsed_big_int TSRMLS_CC)) {
       return 0;
     }
+    *destination = (cass_int32_t) parsed_big_int;
   } else if (Z_TYPE_P(value) == IS_OBJECT &&
              instanceof_function(Z_OBJCE_P(value), php_driver_bigint_ce TSRMLS_CC)) {
     php_driver_numeric *bigint = PHP_DRIVER_GET_NUMERIC(value);
-    *destination = bigint->data.bigint.value;
+    *destination = (cass_int32_t) bigint->data.bigint.value;
   } else {
     throw_invalid_argument(value, param_name, "a long, a double, a numeric string or a " \
                             PHP_DRIVER_NAMESPACE "\\Bigint" TSRMLS_CC);
```

Comment:
Well, the arguments to the Duration constructor are long, double, or numeric string.